agent/src/main/scala/za/co/absa/atum/agent/AtumContext.scala

```diff
@@ -34,18 +36,37 @@ class AtumContext private[agent] (
   val atumPartitions: AtumPartitions,
   val agent: AtumAgent,
   private var measures: Set[Measure] = Set.empty
-) {
+) extends Logging {
 
   def currentMeasures: Set[Measure] = measures
 
   def subPartitionContext(subPartitions: AtumPartitions): AtumContext = {
     agent.getOrCreateAtumSubContext(atumPartitions ++ subPartitions)(this)
   }
 
+  private def validateMeasureApplicability(measure: Measure, df: DataFrame): Unit = {
+    require(
```

Comment:
todo change this to Try - here and below instead of the `log.warn`  as part of this ticket

Reply:
to be done later, I created a ticket for this